### ISSUE TITLE
Case insensitivity and globbing in remapping config that includes zip accessor

### ISSUE THREAD

**Issue by ruzzle** [Velocidex/velociraptor#3597]
At our organisation, we acquire offline collects via velociraptor (live or semi-live) or, if we have no live situation and deal with vm's on ESXi hypervisors (which is actually the case quite often), we use [dissect acquire](https://github.com/fox-it/acquire) (golang is not supported on esxi) to create targeted collections (to zip files).

In our processing pipeline, we aim to use velociraptor (version 0.72.3) to process both dissect acquire and velociraptor offline collects (both zip files) (which is hopefully not too ambitious 😉).

To do so we use velociraptor remappings with configuration snippets like:

```
- type: mount
  description: 'Mount the directory C/* on the C: drive (NTFS)'
  from:
    accessor: zip
    prefix: |
      {
        "DelegateAccessor": "file",
        "DelegatePath": "/path/to/collect.zip",
        "Path": "/fs/C: or /upload/ntfs/..."
      }
  "on":
    accessor: ntfs
    prefix: '\\.\C:'
    path_type: ntfs
```

While testing some artifacts using this remapping configuration, I noticed two issues:
1. Path resolving is case sensitive (in contrast with ntfs)
2. Glob resolving is not supported

Given the issues (see below), I presume it may be related to limitations of the zip accessor? If so, we would very much welcome globbing / case insensitivity support there

### About case insensitivity
I noticed this issue mainly with collects acquired by dissect acquire, as collected files have mixed casing in their paths in the outputted zipfile, e.g. (and notice the windows/system32 folder):
```
fs/C:/windows/system32/winevt/logs
fs/C:/Windows/System32/config/
```
It seems that the lower-cased paths in the zipfile are not supported in the remapping setup, which caused the following errors, e.g. (for the Security event log):
```
[INFO] 2024-07-09T08:23:14Z Unable to open file C:\Windows\System32\winevt\Logs\Security.evtx: Zip: Not found: {"DelegateAccessor":"file","DelegatePath":"/evidence/acquire/DESKTOP-SDN1RPT_20240709074404.zip","Path":"/fs/C:/Windows/System32/winevt/Logs/Security.evtx"}.
```
Though... the security event log is actually in the zip file (but with a mostly lower-cased path)
```
$ unzip -l /evidence/acquire/DESKTOP-SDN1RPT_20240709074404.zip | grep -i security.evtx
2166784  09-19-2020 06:01   fs/C:/windows/system32/winevt/logs/Security.evtx
```

I could verify the case insensitivity was the issue by manually extracting the zip file, changing the casing in the folders, re-creating the zip file and rerunning the processing, which then worked properly.

### About globbing
I noticed this issue in both velociraptor offline collects as well collects acquired by dissect acquire. The globber gave issues during resolving of globs for the NTUSER.DAT files.
```
[INFO] 2024-07-09T07:52:59Z Globber: Zip: Not found: {"DelegateAccessor":"file","DelegatePath":"/evidence/velocollect/Collection-WIN-8OGEKVGFPB7.zip","Path":"/uploads/auto/C%3A/Users/*/NTUSER.DAT"}. while processing HKEY_USERS
```

**Comment by scudette** [comment]
This is an interesting issue since zip filenames **are** case sensitive but we need to treat them as case insensitive. For example we handle this very same problem on Linux by having a special accessor which is case insensitive on Linux https://docs.velociraptor.app/docs/forensic/filesystem/accessors/#file_nocase 

Maybe we need a similar thing for zip or a generic no case accessor that can be applied to any other accessor.

The second issue you mention looks like you are trying to open a file with a literal `*` in it - likely as a result of the `*` being in the remapping config (remappings never use globs but only absolute paths). It is hard to know what the `glob()` was actually running at the time but this is likely to be the case.

**Comment by scudette** [comment]
Should be fixed by #3611
